.github/actions/are-dependent-jobs-successful/action.yml

```diff
@@ -0,0 +1,21 @@
+---
+
+name: are-dependent-jobs-successful
+description: |
+  Evaluates the combined the status results of the provided needs context.
+inputs:
+  needs:
+    required: true
+    description: needs context as JSON string
+outputs:
+  isSuccess:
+    description: the evaluated result of all provided jobs in the needs context.
```

Comment:
I've just found this output name a bit too verbose, if the action name is `are-dependent-jobs-successful` then the output could be `result`, since success is twice in the action and output name.

What do you think?

On the other hand, I've just though the below

if `are-dependent-jobs-successful` can be changed to `check-dependent-jobs` then it could provide more than one output:
- `success`
- `failure`
- `aborted`

and

```
    - id: test
      run: |
        RESULT=$(echo '${{ inputs.needs }}' | jq -s 'map(.[].result) | all(.=="success")')
        echo "success=${RESULT}" >> $GITHUB_OUTPUT
        RESULT=$(echo '${{ inputs.needs }}' | jq -s 'map(.[].result) | all(.=="failure")')
        echo "failure=${RESULT}" >> $GITHUB_OUTPUT
        RESULT=$(echo '${{ inputs.needs }}' | jq -s 'map(.[].result) | all(.=="abort")')
        echo "aborted=${RESULT}" >> $GITHUB_OUTPUT
      shell: bash
```

Do you think that's something we could use in the future by generalising the action to support different status?

Reply:
> I've just found this output name a bit too verbose, if the action name is are-dependent-jobs-successful then the output could be result, since success is twice in the action and output name.

my thinking here was that `isSuccess` is more expressive than `result`. WIth the `is` prefix I already know I can expect a boolean. As of `result` could potentially be anything and as a consumer I would need to look deeper into the docs. 

so if we rename it to `check-dependent-jobs` naming something other than `result` makes sense again.

with your suggestion of haven multiple outputs, I'm worried that `all(.=="failure")` is actually an edge case.
I doubt that this will happen in most cases.

so my suggestion would be:

- rename it to `check-dependent-jobs`
- keep the output name `isSuccess`

and if we see usecases for `isFailure` or `isAborted` when can add it in the future.

Reply:
> so my suggestion would be:
rename it to check-dependent-jobs
keep the output name isSuccess

I like that approach! Thanks again for iterating on this together!

Reply:
https://github.com/elastic/apm-pipeline-library/pull/2126/commits/1659d78dc23d38bf3af0b7c4b20c0c056979e656